### PR TITLE
fix(shared-integration): vscode plug-in stuck

### PR DESCRIPTION
**Pull request by Simon-He95**
close: #3711

The original regular expression would cause an error when matching code. Now it has been updated and the ending tag is filtered.

## origin:

image

image

## new:

image

### CODE REVIEW

**Comment by zcf0508** [comment]
image

**Comment by Simon-He95** [comment]
> image
You can take a look to see if there are any problems with this regular rule.
`/<[^\/](?:[^<>\[\]"]+)(?:>|"[^"]*"(?:(?:[^<>\[\]{]+)(?:>|{[^}]*}|(?:\[[^\]]*\])))*|(?:\[[^\]]*\](?:(?:[^<>\[\]{"]+)(?:>|"[^"]*"|{[^}]*}|(?:\[[^\]]*\])))*))/g`

File: packages/shared-integration/src/defaults.ts [hunk 1]
```diff
@@ -15,8 +15,8 @@ export const defaultFilesystemGlobs = [
 export const defaultIdeMatchInclude: RegExp[] = [
   // String literals
   /(['"`])[^\1]*?\1/g,
-  // HTML tags ignore [>]
-  /<(?:[^<>\[\]]+|\[[^\]]*\])*>/g,
+  // HTML tags
+  /<[^\/](?:[^<>\[\]"]+)(?:>|"[^"]*"(?:(?:[^<>\[\]{]+)(?:>|{[^}]*}|(?:\[[^\]]*\])))*|(?:\[[^\]]*\](?:(?:[^<>\[\]{"]+)(?:>|"[^"]*"|{[^}]*}|(?:\[[^\]]*\])))*))/g,
```
Comment (antfu) [inline packages/shared-integration/src/defaults.ts:19]: This seems to be over complicated to me

Maybe:

```suggestion
  /<(?:"[^"]*"|'[^"]*'|[^>])+>/g,
```

Comment (Simon-He95) [reply packages/shared-integration/src/defaults.ts]: image
He will miss some needed matches

**Comment by Simon-He95** [comment]
> image
> 
> image

This is an irregular way of writing. You should not write similar code directly in attributes. In addition, you should not use words such as relative when naming.

**Pull request merged**